core/services/relay/relay.go

```diff
@@ -88,3 +95,28 @@ func (r *relayerAdapter) HealthReport() map[string]error {
 	maps.Copy(r.RelayerExt.HealthReport(), hr)
 	return hr
 }
+
+type ErrLogFiltersNotSupported struct {
+	relayName string
+}
+
+func (e ErrLogFiltersNotSupported) Error() string {
+	return fmt.Sprintf("Log filtering is not supported by relay %s", e.relayName)
+}
+
+func (r *relayerAdapter) RegisterLogFilters(rargs types.RelayArgs, q pg.Queryer) (err error) {
+	relay, ok := r.Relayer.(LogPollerCapable)
+	if !ok {
+		return ErrLogFiltersNotSupported{r.Relayer.Name()} // only supported for evm currently
+	}
+	return relay.RegisterLogFilters(rargs, q)
+
+}
+
+func (r *relayerAdapter) UnregisterLogFilters(rargs types.RelayArgs, q pg.Queryer) (err error) {
+	relay, ok := r.Relayer.(LogPollerCapable)
+	if !ok {
+		return ErrLogFiltersNotSupported{r.Relayer.Name()} // only supported for evm currently
+	}
+	return relay.UnregisterLogFilters(rargs, q)
+}
```

Comment:
The relayer provider methods are where register filter is called on develop branch presently.  The goal of this PR is to move all calls to RegisterFilter out of those methods and into `OnCreateJob()`... so I think the second part of your question amounts to "what's the motivation for making this change?"  The difference is that the provider methods are called every time the node starts up, or when a new job service is being spawned.  Whereas `OnCreateJob()` is called atomically along with the actual job creation.

In the case of `UnregisterFilter()` it's mandatory that this not happen every time the node shuts down--removing the filters while the node is down can lead to missed logs.

In the case of `RegisterFilter()`, it's less essential but has a some benefits.  The main benefit is that if there is any problem with the job spec, it can let the user know as soon as they try to create the job.  If `RegisterFilter()` isn't called until later on when the services are being spawned, it's too late... the job already exists in the db, it will just fail to start up.  So the user has created an invalid job but then has to check the results of the first job run to see that it's failing.  Then they have to go back and delete the job and recreate it.  Another benefit is that it just makes sense for `RegisterFilter()` and `UnregisterFilter()` to be symmetric, one called only when jobs are created and one called only when jobs are deleted--cleaner API and less confusing for developers.

> Does LogFilter need to be part of the relayer API? Rather than an implementation detail? 

That's a fair question... I suppose this doesn't necessarily have to be LogPoller specific.  I guess instead of `RegisterLogFilters()` and `UnregisterLogFilters()` we could have more general relay methods called `OnCreateJob()` and `OnDeleteJob()`.  They could include anything relay-specific that has to happen at job creation or job deletion time... which could call `RegisterLogFilters()` and `UnregisterLogFilters()` in the case of evm, but either something else or nothing in the case of non-evm blockchains.  What do you think?

Reply:
> This luxury will not be available in LOOP Plugin mode.

Yikes... that's a good point, I'm not sure how we'll deal with that.  We may have to relax the atomicity requirements, and instead add some kind of cleanup process that fixes up any inconsistencies.  I guess we can cross that bridge when we come to it?

Reply:
If it is clear how to relax that assumption now then it would be good to think ahead. Splitting the schemas is likely only a few weeks away.